controllers/keystoneapi_controller.go

```diff
@@ -113,7 +113,12 @@ func (r *KeystoneAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	// initialize status
 	//
 	if instance.Status.Conditions == nil {
-		instance.Status.Conditions = condition.List{}
+		instance.Status.Conditions = condition.Conditions{}
+		instance.Status.Conditions.Init(nil)
+		// Register overall status immediately to have an early feedback e.g. in the cli
```

Comment:
This also helps avoiding that a later r.Update() call resets the instance.Status field. Probably we need to call r.Status().Update after every condition change to avoid that behavior

Reply:
ack. if we set a condition then return, that will trigger the defer and persist the condition. So the only problematic scenario when we set a success condition, continue the current reconcile loop and call r.Update() before we return or before r.Status.Update() is called.